pyqtgraph/graphicsItems/PlotCurveItem.py

```diff
@@ -1,20 +1,136 @@
-from ..Qt import QtCore, QtGui, QtWidgets
+from ..Qt import QtCore, QtGui, QtWidgets, QT_LIB
 
-HAVE_OPENGL = hasattr(QtWidgets, 'QOpenGLWidget')
+import importlib
 import math
-import sys
 import warnings
 
 import numpy as np
 
 from .. import Qt, debug
 from .. import functions as fn
 from .. import getConfigOption
+from ..Qt import OpenGLConstants as GLC
 from .GraphicsObject import GraphicsObject
 
 __all__ = ['PlotCurveItem']
 
 
+class OpenGLState:
```

Comment:
This _may_ need to inherit from `QObject` due to the `cleanup` method being a `Slot`.

In some non-pyqtgraph Qt related work, I can into issues when connecting to a slot that wasn't decorated and wasn't decorated with the `@Slot()` decorator.

https://www.riverbankcomputing.com/static/Docs/PyQt6/signals_slots.html#the-pyqtslot-decorator

> Connecting a signal to a decorated Python method also has the advantage of reducing the amount of memory used and is slightly faster.

I know pyqtgraph doesn't decorate its slots but I anticipate this being a change that we'll roll out in the near future.

Reply:
Was your issue because of being called from another thread?
In this case here, cleanup has to be a direct call and thus doesn't need to be registered as a QObject metamethod.